alpha/lib/model/category.php

```diff
@@ -491,9 +489,21 @@ public function setDeletedAt($v, $moveEntriesToParentCategory = false)
 	public function deleteChildCategories()
 	{
 		$fullIds = $this->getFullIds();
-		$categoriesIds = self::getCategoriesStartWithIds($fullIds.'>');
-		self::updateDeletedCategoriesInDB($categoriesIds);
-		$categories = self::getCategories($categoriesIds);
+		$categoriesIds = $this->getDescendantCategoriesIds();
+
+		if (isset($categoriesIds) && !empty($categoriesIds))
+		{
+			$update = KalturaCriteria::create(categoryPeer::OM_CLASS);
+			$update->add(categoryPeer::DELETED_AT, time());
+			$update->add(categoryPeer::UPDATED_AT, time());
+			$update->add(categoryPeer::STATUS, CategoryStatus::DELETED);
+			$update->add(categoryPeer::ID, $categoriesIds, KalturaCriteria::IN);
+			categoryPeer::doUpdate($update);
+		}
+
+		categoryPeer::setUseCriteriaFilter(false);
+		$categories = categoryPeer::retrieveByPKs($categoriesIds);
+		categoryPeer::setUseCriteriaFilter(false);
 
 		foreach ($categories as $categoryToDelete) {
 			kEventsManager::raiseEvent(new kObjectDeletedEvent($categoryToDelete));
```

Comment:
Please set the status, deleted-at and the updated-at again (without saving) before reindexing, because the objects could be loaded from the cache.